Automattic/wp-calypso
# Themes: show auto loading homepage modal when switching to a TFT

**User Story:**
As a dot com user, I'd like to be warned if the homepage of my site will be replaced by the theme when I switch it.

#### Changes proposed in this Pull Request

This PR shows a warning modal when the user switches to a template-first theme (TFT) in order to let them know that the homepage will be replaced by a new one, and the current one will be set ad draft. 

#### Testing instructions

Test switching to a TFT and confirming you see the warning modal, similar to the following:

<img src="https://user-images.githubusercontent.com/77539/74665448-7fcb9900-517e-11ea-95a2-95eab2a01d2c.png" width="500px" />

Also, confirm that the modal won't appear for not TFT themes, activating the theme and immediately showing the activated modal:

<img src="https://user-images.githubusercontent.com/77539/74665625-d5a04100-517e-11ea-8343-d065652f52f5.png" width="500px" />

This behavior should be tested from different pages:

* Themes list: `http://calypso.localhost:3000/themes/<your-testing-site>`
<img width="500" alt="Screen Shot 2020-02-17 at 1 00 08 PM" src="https://user-images.githubusercontent.com/77539/74669332-b35df180-5185-11ea-8fa7-5729c675e9a3.png">

* Theme info: `http://calypso.localhost:3000/theme/<theme>/<your-testing-site>`
<img width="500" alt="Screen Shot 2020-02-17 at 1 00 52 PM" src="https://user-images.githubusercontent.com/77539/74669318-ae00a700-5185-11ea-9e63-a1c8f8151159.png" />





Fixes #39073
